app.rb

```diff
@@ -0,0 +1,187 @@
+      books.each_with_index { |book, idx| puts "#{idx}) Title: #{book.title}, Author: #{book.author}" }
+    else
+      if books.empty?
+        puts "No books added yet\n\n"
+        return
+      end
+      books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    end
+    puts ''
+  end
+
+  def list_people(people, with_id: false)
+    if with_id
+      people.each_with_index do |person, idx|
+        puts "#{idx}) [#{person.class}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+      end
+    else
+      if people.empty?
+        puts "No person created yet\n\n"
+        return
+      end
+      people.each { |person| puts "[#{person.class}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}" }
+    end
+    puts ''
+  end
+
+  def add_rental(books, people) # rubocop:disable Metrics/MethodLength
+    if people.empty? || books.empty?
+      puts "Either People or book List is empty. Please populate these lists in order to check rentals\n\n"
+      return
+    end
+
+    puts 'Select a book from the following list:'
+    list_books(books, with_id: true)
+    book_idx = UserInput.get_number('Book: ', 'Please enter a valid number!')
+
+    puts 'Select a person from the following list:'
+    list_people(people, with_id: true)
+    person_idx = UserInput.get_number('Borrower: ', 'Please enter a valid number!')
+    date = UserInput.get_date('Date: ', 'Please enter a valid date!')
+
+    begin
+      person = people.at(person_idx)
+      book = books.at(book_idx)
+      rental = Rental.new(date, book, person)
+      person.rentals << rental
+    rescue NoMethodError
+      puts "Book or person not found. Please try again!\n\n"
+      return add_rental(books, people)
+    end
+
+    puts "Rental Added!\n\n"
+  end
+
+  def list_rentals(people)
+    if people.empty?
+      puts "People List is empty. Please add people in order to check their rentals\n\n"
+      return
+    end
+    puts 'Select a person ID from the following list:'
+    list_people(people)
+
+    person_id = UserInput.get_number('Person ID: ', 'Please enter a valid number!')
+    person = people.find { |current_person| current_person.id == person_id }
+    if person.nil?
+      puts "Person with ID: #{person_id} not found!\n\n"
+      return list_rentals(people)
+    end
+
+    if person.rentals.empty?
+      puts "#{person.name} didn't rent any book yet\n\n"
+    else
+      puts "Rentals for #{person.name}:"
+      person.rentals.each { |rental| display_rental(rental) }
+      puts "\n"
+    end
+  end
+
+  def display_rental(rental)
+    case rental.book
+    when Book
+      puts "Date: #{rental.date}, Book: #{rental.book.title} By #{rental.book.author}"
+    when Teacher
+      puts "Date: #{rental.date}, Teacher: #{rental.book.name}, Specialization: #{rental.book.specialization}"
+    when Student
+      puts "Date: #{rental.date}, Student: #{rental.book.name}, Parent Permission: #{rental.book.parent_permission}"
+    end
+  end
+end
```

Comment:
- [ ] Great job working on the UI so far but I think you have a small problem here when displaying a rental with an ID so I think it would be great if you fix this issue because I tried it many times after creating a rental but I still can't be able to see it and it shows me this error message ⬇ so kindly take a look and try to solve the problem here 🚀💯🔥🙏🏻

## in `display_rental': undefined method `parent_permission' 

![image](https://github.com/Caren-Koroeny/school-library/assets/58553711/1068f0d7-c450-4400-98fb-98f6a4e35b67)